pkg/prometheus/promcfg.go

```diff
@@ -421,7 +422,13 @@ func compareScrapeTimeoutToScrapeInterval(scrapeTimeout, scrapeInterval monitori
 
 // Generate creates a serialized YAML representation of a Prometheus configuration using the provided resources.
 func (cg *ConfigGenerator) Generate(
-	p *v1.Prometheus,
+	evaluationInterval v1.Duration,
+	queryLogFile string,
+	ruleSelector *metav1.LabelSelector,
+	exemplars *v1.Exemplars,
+	tsdb v1.TSDBSpec,
+	alerting *v1.AlertingSpec,
+	remoteRead []v1.RemoteReadSpec,
```

Comment:
non-blocking comment, doing it just in case someone has a good solution: I'm a bit afraid that passing the non-agent Prometheus Config like this might become a bit annoying to maintain 🤔 not sure if putting these inside of a struct might make sense.

Reply:
I think this is still manageable but I can definitely see your point! My preference here would be to do what the current maintainers prefer, so... wdyt @prometheus-operator/prometheus-operator-reviewers?

Reply:
This is fine by me. When we move to implementing the Prometheus agent part, I imagine that we'll rename `*ConfigGenerator.Generate()` to `*ConfigGenerator.GenerateServerConfiguration()` and create a `*ConfigGenerator.GenerateAgentConfiguration()` method